FMS/doc/man/man3/chGrp.3
'\" t
.\"     Title: chGrp
.\"    Author:  SysFera SA <info@sysfera.com>
.\" Generator: DocBook XSL Stylesheets v1.77.1 <http://docbook.sf.net/>
.\"      Date: December 2012
.\"    Manual: FMS C++ API Reference
.\"    Source: VISHNU ZMQ_beta
.\"  Language: English
.\"
.TH "CHGRP" "3" "December 2012" "VISHNU ZMQ_beta" "FMS C++ API Reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
chGrp \- changes group owner of remote files/directories\&.
.SH "SYNOPSIS"
.HP \w'int\ vishnu::chGrp('u
.BI "int vishnu::chGrp(const\ string&\ " "sessionKey" ", const\ string&\ " "group" ", const\ string&\ " "path" ");"
.SH "DESCRIPTION"
.PP
Changes the group attribute of a file or directory at the location given by the path parameter\&. The path must be provided in the form \*(Aqmachine_id:path\*(Aq (the machine_id values can be obtained using the vishnu_list_machine command)
.SH "ARGUMENTS"
.PP
\fIsessionKey\fR
.RS 4
Input argument\&. The session key\&.
.RE
.PP
\fIgroup\fR
.RS 4
Input argument\&. The new group owner of file/directory\&.
.RE
.PP
\fIpath\fR
.RS 4
Input argument\&. The file/directory following the pattern [host:]file path\&.
.RE
.SH "EXCEPTIONS"
.PP
The following exceptions may be thrown:
.PP
"Vishnu not available (Service bus failure)" [1]
.RS 4
.RE
.PP
"Vishnu not available (Database error)" [2]
.RS 4
.RE
.PP
"Internal Error: Undefined exception" [9]
.RS 4
.RE
.PP
"an option or a parameter provided is invalid for this service" [10]
.RS 4
.RE
.PP
"Undefined configuration parameter" [12]
.RS 4
.RE
.PP
"The user is not an administrator" [25]
.RS 4
.RE
.PP
"The session key is unrecognized" [28]
.RS 4
.RE
.PP
"The session key has expired\&. The session is closed\&." [29]
.RS 4
.RE
.PP
"The machine id is unknown" [32]
.RS 4
.RE
.PP
"The local account is unknown" [38]
.RS 4
.RE
.PP
"The path provided is invalid\&." [201]
.RS 4
.RE
.PP
"Runtime error" [202]
.RS 4
.RE
.PP
"The transfer id is unknown" [203]
.RS 4
.RE
.SH "AUTHOR"
.PP
\fB SysFera SA\fR <\&info@sysfera.com\&>
.RS 4
Developed the first version of these man pages as part of the VISHNU project.
.RE
.SH "COPYRIGHT"
.br
Copyright \(co 2012 SysFera SA
.br
.PP
These manual pages are provided under the following conditions:
.PP
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
.RE
.PP
This software is governed by the CECILL licence under French law and abiding by the rules of distribution of free software. You can use, modify and/ or redistribute the software under the terms of the CeCILL license as circulated by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
.PP
As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license, users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the successive licensors have only limited liability.
.PP
In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or developing or reproducing the software by the user in light of its specific status of free software, that may mean that it is complicated to manipulate, and that also therefore means that it is reserved for developers and experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the software's suitability as regards their requirements in conditions enabling the security of their systems and/or data to be ensured and, more generally, to use and operate it in the same conditions as regards security.
.sp


ChangeLog
+-----------------------------+
| VISHNU 3.0.0 - Beta version |
+-----------------------------+

* Fixed bugzilla #341 - Rename of FMS commands using POSIX name
* WS do not work

+-----------------------------+
| VISHNU 2.0.0 - Beta3        |
+-----------------------------+
* Improve the generation of the ID for vishnu objects
* Add documentation for admin with distant databases
* Clean unused files
* Fixed bug 440
* Fix LSF bug for -N and -P options in submit
* Fix bad types in comparison from unsigned int to size_t
* Add PBS_Pro support
* Add specific parameters for batchs schedulers (option -S with vishnu_submit_job)
* Add default parameters in config files for batchs
* WS do not work

+-----------------------------+
| VISHNU 2.0.0 - Beta2        |
+-----------------------------+

* Fixed bugzilla #417 - compilation problems with LoadLeveler
* Fixed bugzilla #410 - solve restoreConfiguration service problems when webBoard data have been saved in the database
* Fixed bugzilla #431 - vishnu_get_job_output renvoie une erreur Dagda
* Fixed bugzilla #426 alias #433 - vishnu_list_jobs with a list of statuses failed (vishnu_list_job -S)
* Fixed bugzilla #434 - vishnu_get_job_output throws core dump exception when the error path or the output do not longer exist
* Fixed error to try to reconnect when having mysql error server has gone
* Changed - input files are now stored in the local account's home directory

+-----------------------------+
| VISHNU 2.0.0 - Beta version |
+-----------------------------+

* Added status display on textual form for vishnu_list_users and vishnu_list_machines 
* Added Client/Server version management for services: connect, reconnect and changePassword
* Added the use of work identifier during job submission
* Added job submission with textual parameters
* Added local file used on the submission script
* Added the possibility to get files generated by the submission script
* Added support for SGE batch scheduler
* Fix mac compilation problem for the client. Now use libcrypto instead of libcrypt
* Change Python API to use real python list
* Fixed bugzilla #369 - Error in FMS list dir if space in name
* Fixed bugzilla #399 - Extended vishnu_list_jobs options for listing all information about job

+--------------+
| VISHNU 1.3.1 |
+--------------+

* Fixed bugzilla #363 - Automatic user ssh key configuration on a remote machine 
* Fixed bugzilla #368 - Lack of ListUserOptions class on Python interface
* Fixed compilation problem where libldap was not linked to the sed
* Fixed bugzilla #350 - Clients of FMS module can be out of DNS for copy and move of files

+--------------+
| VISHNU 1.3.0 |
+--------------+

* Added LDAP support for users authentication
* Added support for LSF batch scheduler
* Fixed bugzilla #335 - Problem with big file transfer 
* Fixed bugzilla #343 - Added batch job identifier on job structure

+--------------+
| VISHNU 1.2.7 |
+--------------+

* Added service for automatically submit a job

* Added keyword autom as machine id to submit job API function and Command
line to allow automatic submission

* Added service for list all submitted jobs through VISHNU on all machines

* Added keyword all as machine id to list jobs API function and Command
line to allow the listing of jobs on all machines

* Extended submit job options: added selectQueueAutom and loadType options

* Extended list jobs options: added multipleStatus and batchJob options 

* Fixed bugzilla #326 where with the option -h of connect command did not work 

* Fixed bugzilla #337 : Problem with boost installed in user space

+--------------+
| VISHNU 1.2.6 |
+--------------+

* Extended submit job options: added mailNotification, mailNotifyUser,
group, workingDir cpuTime options

* Added JOB OUTPUT ENVIRONMENT VARIABLES

* Added the use of .netrc file for the connection

* Added recursive option for the creation of the directory

* Modified listDir C++ and Python API: new structure added 

* Added -w option for the vishnu_reconnect function

* Added a new command line vishnu_current_session_id added

+--------------+
| VISHNU 1.2.5 |
+--------------+

* Modified python API of the function connect/reconnect

* Fixed bugzilla #252 where with the MYSQL Database no error is returned 
by the function setUpdateFrequence if a negative value is given

* Fixed bugzilla #253 where the getProcesses returns an empty list
while there is running vishnu processes

* Fixed bugzilla #255 where the submitted jobs under LoadLeveler
batch scheduler stays in status RUNNING

* Fixed bugzilla #261 where vishnu_list_jobs filtered by 
queue name returns [this queue name is unknown] while 
this queue exists.

* Fixed bugzilla #262 there was a mismatch on the jobs state through web
service interface

* Fixed bugzilla #263 where the value of the nbQueue parameter is not 
returned through web service interface.  

* Fixed bugzilla #264 where vishnu_get_job_info reverses
the jobOutputPath and jobErrorPath

* Fixed bugzilla #266 where in vishnu_update_user the 
option -s was missing

* Fixed bugzilla #268 where the compilation failed on ubuntu 11.10

* Updated of CMakeList.txt for automatically set default value of the
datase library.

* Fixed bugzilla #280 where the umssed segfault on changePassword with userId empty 

+--------------+
| VISHNU 1.2.4 |
+--------------+

* updated of the documentation
* modified python API of the function connect/reconnect

+--------------+
| VISHNU 1.2.3 |
+--------------+

* Added support for SLURM batch scheduler

+--------------+
| VISHNU 1.2.2 |
+--------------+

* Reduce cli code duplication
* Fix audit IMS and FMS
* Fix bugs

+--------------+
| VISHNU 1.2.1 |
+--------------+

* Add the machineid as a forced parameter
* Fix TMS code quality (audit correction)
* Remove getHostname dependency

+--------------+
| VISHNU 1.2.0 |
+--------------+

* Add FMS et IMS modules first versions.
* Python and WS api available for FMS and IMS

+--------------+
| VISHNU 1.1.0 |

FMS/src/cli/chgrp.cpp
/**
 * \file chgrp.cpp
 * This file defines the VISHNU chgrp command
 * \author Daouda Traore (daouda.traore@sysfera.com)
 */


#include "CLICmd.hpp"
#include "utilVishnu.hpp"
#include "cliError.hpp"
#include "cliUtil.hpp"
#include "api_ums.hpp"
#include "api_fms.hpp"
#include "sessionUtils.hpp"
#include "cmdArgs.hpp"
#include <boost/bind.hpp>
#include "GenericCli.hpp"

namespace po = boost::program_options;

using namespace std;
using namespace vishnu;

struct ChangeFileGroupFunc {

  std::string mpath;
  std::string mgroup;

  ChangeFileGroupFunc(const std::string& path,const std::string& group):mpath(path),mgroup(group){}

  int operator()(std::string sessionKey) {

      int res =chgrp(sessionKey, mgroup, mpath);
    return res;
  }
};


int main (int argc, char* argv[]){

  /******* Parsed value containers ****************/
  string dietConfig;
  string path;
  string group;

  /**************** Describe options *************/
  boost::shared_ptr<Options> opt=processOpt(argv[0], dietConfig);

  opt->add("group,g",
      "The new group owner of file/directory",
      HIDDEN,
      group,1);
  opt->setPosition("group",1);

  opt->add("path,p",
      "The file/directory following the pattern [host:]file path",
      HIDDEN,
      path,1);
  opt->setPosition("path",1);

  bool isEmpty;
  GenericCli().processListOpt( opt, isEmpty,argc,argv,"group path");
  ChangeFileGroupFunc apiFunc(path,group);
  return GenericCli().run(apiFunc, dietConfig, argc, argv);

}
